src/main/java/com/github/firmwehr/gentle/firm/optimization/Optimizer.java

```diff
@@ -0,0 +1,46 @@
+package com.github.firmwehr.gentle.firm.optimization;
+
+
+import firm.Graph;
+import firm.Program;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Optimizer {
+	private final List<GraphOptimizationStep> graphOptimizationSteps;
+
+	private Optimizer(List<GraphOptimizationStep> graphOptimizationSteps) {
+		this.graphOptimizationSteps = List.copyOf(graphOptimizationSteps);
+	}
+
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	public void optimize() {
+		for (Graph graph : Program.getGraphs()) {
+			boolean changed;
+			do {
+				changed = false;
+				for (GraphOptimizationStep step : this.graphOptimizationSteps) {
+					changed |= step.optimize(graph);
```

Comment:
If the first step changes the graph, the whole pipeline is executed twice. That might not be needed, but is probably fine for now.